src/maker/server.rs

```diff
@@ -224,28 +224,66 @@ fn setup_fidelity_bond(maker: &Arc<Maker>, maker_address: &str) -> Result<(), Ma
                 Err(e) => {
+
                     if let WalletError::InsufficientFund {
                         available,
                         required,
                     } = e
                     {
-                        log::warn!("Insufficient fund to create fidelity bond.");
-                        let amount = required - available;
-                        let (_, addr, _) = maker
-                            .get_wallet()
-                            .read()?
-                            .get_next_fidelity_address(locktime)?;
-                        log::info!("Send {} sats to {}", amount, addr);
-                        if cfg!(feature = "integration-test") {
-                            sleep(Duration::from_secs(3));
-                        } else {
-                            sleep(Duration::from_secs(300)); // Wait for 5 mins in production
+                        
+                        loop {
+
+                            let (_, addr, _) = maker
+                                .get_wallet()
+                                .read()?
+                                .get_next_fidelity_address(locktime)?;
+
+                            let incoming_utxo = maker
+                                .get_wallet()
+                                .read()?
+                                .rpc
+                                .list_unspent(Some(0), Some(9999999), Some(&[&addr]), None, None)
+                                .map_err(|e| WalletError::Rpc(e))?;
+
+                            log::info!("incomming utxo: {:?}", incoming_utxo);
+                            log::info!("balance: {:?}", maker.get_wallet().read()?.balance());
+                            
+                            //  No transaction is pointing to this address
+                            if incoming_utxo.is_empty() {
+                                log::warn!("Insufficient fund to create fidelity bond.");
+
+                                let amount = required - available;
+                                log::info!("Send {} sats to {}", amount, addr);
+                            }
+                            
+                            // Tx funding to that address is created.
+                            else {
+                                let utxo = incoming_utxo[0].clone();
+
+                                if utxo.confirmations == 0 {
+                                    log::info!("Transaction: {:?} seen in mempool, waiting for confirmation.", utxo.txid);
+                                } else {
+                                    //  Tx is confirmed.
+                                    log::info!(
+                                        "Transaction: {:?} confirmed at blockheight:",
+                                        utxo.txid
+                                    );
+                                    break;
+                                }
+                            }
```

Comment:
Lets disable this code for now and only put the sync call. Will be easier to debug.